### PR TITLE
MGDOBR-219 - Enforce SSL Connections from Fleet Manager to Database

### PR DESCRIPTION
**Pull request by msmagnanijr**
Signed-off-by: Mauricio Magnani <msmagnanijr@gmail.com>

[MGDOBR-219](https://issues.redhat.com/browse/MGDOBR-219)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by msmagnanijr** [comment]
@all the certificate will be available as both, environment variables and volume.

File: app-interface/manifests/rhose-fleet-manager.yml [hunk 1]
```diff
@@ -201,6 +206,13 @@ objects:
...
                       name: ${FLEET_MANAGER_SECRET_NAME}
                       key: SHARD_ID
+              volumeMounts:
+              - name: rds
+                mountPath: /secrets/rds
+          volumes:
+          - name: rds
+            secret:
+              secretName: ${RDS_SECRET_NAME}
               ports:
```
Comment (sutaakar) [inline app-interface/manifests/rhose-fleet-manager.yml:216]: Is it properly aligned? I would expect the content below to be part of `containers` block, not `volumes`

Comment (msmagnanijr) [reply app-interface/manifests/rhose-fleet-manager.yml:216]: @sutaakar looks ok but I'll check it out.


```
 oc apply -f app-interface/manifests/rhose-fleet-manager.yml --dry-run
W0516 12:59:08.669021   83255 helpers.go:555] --dry-run is deprecated and can be replaced with --dry-run=client.
W0516 12:59:09.551338   83255 shim_kubectl.go:55] Using non-groupfied API resources is deprecated and will be removed in a future release, update apiVersion to "template.openshift.io/v1" for your resource
template.template.openshift.io/rhose-fleet-manager created (dry run)
```

Comment (msmagnanijr) [reply app-interface/manifests/rhose-fleet-manager.yml:216]: @sutaakar https://medium.com/avmconsulting-blog/secrets-management-in-kubernetes-378cbf8171d0  I think it's ok, right? In this case I'm wanting to mount the secret as a volume.

Comment (sutaakar) [reply app-interface/manifests/rhose-fleet-manager.yml:216]: I am referring to this block:
```
          volumes:
          - name: rds
            secret:
              secretName: ${RDS_SECRET_NAME}
              ports:
                - containerPort: 8080
              livenessProbe:
                httpGet:
                  path: /q/health/live
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
              readinessProbe:
                httpGet:
                  path: /q/health/ready
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
```
When you apply the template then ports and probes are gone.

Comment (msmagnanijr) [reply app-interface/manifests/rhose-fleet-manager.yml:216]: @sutaakar got it! tks

File: app-interface/manifests/rhose-fleet-manager.yml [hunk 2]
```diff
@@ -201,20 +206,27 @@ objects:
...
+              path: /q/health/live
+              port: 8080
+            initialDelaySeconds: 10
+            periodSeconds: 10
+          readinessProbe:
+            httpGet:
+              path: /q/health/ready
+              port: 8080
+            initialDelaySeconds: 10
+            periodSeconds: 10
```
Comment (sutaakar) [inline app-interface/manifests/rhose-fleet-manager.yml:229]:
```suggestion
              ports:
                - containerPort: 8080
              livenessProbe:
                httpGet:
                  path: /q/health/live
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
              readinessProbe:
                httpGet:
                  path: /q/health/ready
                  port: 8080
                initialDelaySeconds: 10
                periodSeconds: 10
              volumeMounts:
              - name: rds
                mountPath: /secrets/rds
          volumes:
          - name: rds
            secret:
              secretName: ${RDS_SECRET_NAME}
```

Comment (sutaakar) [reply app-interface/manifests/rhose-fleet-manager.yml:229]: Basically to keep what was available before, adding `volumes` below

**Review by sutaakar — APPROVED** [review]

**Pull request merged**